spring-batch-docs/asciidoc/common-patterns.adoc

```diff
@@ -531,7 +531,7 @@ public Trade read() throws Exception {
             return t; // Record must end with footer
         }
     }
-    Assert.isNull(t, "No 'END' was found.");
+    Assert.notNull(t, "No 'END' was found.");
```

Comment:
To be consistent, we need to update the code accordingly. But when I apply this change to `MultiLineTradeItemReader`, an `IllegalArgumentException` is thrown in `org.springframework.batch.sample.iosample.MultiLineFunctionalTests#testJob` (the test still passes though..). This is out of scope of this PR, so please revert this change for now. The rest LGTM.